### ISSUE TITLE
RHCOS installation for OCP 4.3 fails at first boot

### ISSUE THREAD

**Issue by rahulsingh45-tech** [RedHatOfficial/ocp4-helpernode#55]
Hi Team,

While booting RHCOS for OCP 4.3 machine, the system shows messages as follows and boot is getting failed with following errors:

can't open InitiatorName configuration file /etc/iscsi/initiatorname.iscsi
iscsid[606]: Warning: InitiatorName file /etc/iscsi/initiatorname.iscsi does not exist or does not contain a properly formatted InitiatorName. 
iscsid[606]: can't open InitiatorAlias configuration file /etc/iscsi/initiatorname.iscsi

dracut-initqueue[271]: Warning: dracut-initqueue timeout - starting timeout scripts
dracut-initqueue[271]: Warning: dracut-initqueue timeout - starting timeout scripts
dracut-initqueue[271]: Warning: dracut-initqueue timeout - starting timeout scripts
dracut-initqueue[271]: Warning: dracut-initqueue timeout - starting timeout scripts

**Comment by christianh814** [comment]
That could be any number of things. Which method did you use? DHCP or Static?

**Comment by rahulsingh45-tech** [comment]
Hi Christian,

We are trying with Static ip method. Kindly help us on why we are getting these errors and boot is getting failed.

**Comment by christianh814** [comment]
Since you are using the static ip method, the issue could be that you're not providing the right disk. Double check and make sure that `coreos.inst.install_dev` is pointing to the right device. This will vary from environment to environment. It could be sda, vda, or even nvme0n1p1. It will depend on your environment.

Also, Make sure the interface is set correctly on the `ip=` kernel option as well. It could be `enp1s0`, `eth0` or `ens3`. It all depends.

Make sure you double check these before typing them in.

**Comment by rahulsingh45-tech** [comment]
Thanks Christian for the response.
We had tried sda and vda both.  How we can verify whether its pointing to right device or not ?

We are using below parameter in kernel command.

coreos.inst.install_dev=sda coreos.inst.image_url=http://10.254.9.60:80/install/rhcos-4.3.8-x86_64-metal.x86_64.raw.gz
coreos.inst.ignition_url=http://10.254.9.60:80/ignition/bootstrap.ign
ip=10.254.9.62::10.254.8.1:255.255.254.0:rhcos01.bacadgsl.com:enp1s0:none:10.254.9.60

**Comment by christianh814** [comment]
> ip=10.254.9.62::10.254.8.1:255.255.254.0:rhcos01.bacadgsl.com:enp1s0:none:10.254.9.60

This is incorrect. You need to add `nameserver` for your DNS server. This is probably the issue.

```
ip=10.254.9.62::10.254.8.1:255.255.254.0:rhcos01.bacadgsl.com:enp1s0:none nameserver=10.254.9.60
```

> We had tried sda and vda both. How we can verify whether its pointing to right device or not ?

The easiest way is to boot a RHEL 8 install and run `fdisk -l` on your VM/Server to find out . Also double the check the interface with `ip a`

**Comment by christianh814** [comment]
So I just noticed something...

```
255.255.254.0
```

Unfortunately, the playbook doesn't support anything other than a `/24` right now #2 

You can possibly get it working by removing the reverse file `/var/named/reverse.db` and changing the `named.conf` to not include a reverse. I would try also installing 4.4.3 since it relys less on DNS than 4.3.x did.

If you can't switch to a `/24`, then you may need to manually edit DNS to comply with your environment.

**Comment by rahulsingh45-tech** [comment]
Hi Christian,

Previous error is resolved and boot is completed after using eth0 in kernel command.
After restarting the VM , we are getting login Window where need to provide login and password. Can you please help on what should be the login user for rhcos machine and also after few minute we are getting below errors. Also attached the screenshot for the same here.

SELinux: mount invalid.  Same superblock, different security settings for (dev mqueue, type mqueue)"


![rhcos_Error](https://user-images.githubusercontent.com/64540939/81821390-694f9680-954f-11ea-9143-d090cfd930a8.jpg)

**Comment by christianh814** [comment]
Those are normal errors that you can safely ignore.

You can login to the system only via ssh. Use the key you provided in the `install-config.yaml` file.

```
ssh -i <ssh key> <fqdn of node>
```

**Comment by rahulsingh45-tech** [comment]
Tried to connect via ssh in helper node. getting below error.

[root@BACADGSL etc]# ssh -i /home/bacauser/.ssh/id_rsa.pub ocp4.bacadgsl.com
ssh: Could not resolve hostname ocp4.bacadgsl.com.ocp4.bacadgsl.com: Name or service not known
[root@BACADGSL etc]# ssh core@10.254.9.62
ssh: Could not resolve hostname 10.254.9.62.ocp4.bacadgsl.com: Name or service not known
[root@BACADGSL etc]#

**Comment by christianh814** [comment]
Try it without the fqdn or modify your `~/.ssh/config` file

**Comment by rahulsingh45-tech** [comment]
Can you please help with the ideal setting for /.ssh/config file.

**Comment by christianh814** [comment]
The playbook should have set it correctly.

it will all depend on what your dns is for your bootstrap/workers/masters.

for example, if your vars.yaml file looked like this...

```
helper:
  name: "helper"
  ipaddr: "192.168.7.77"
dns:
  domain: "example.com"
  clusterid: "ocp4"
  forwarder1: "8.8.8.8"
  forwarder2: "8.8.4.4"
bootstrap:
  name: "bootstrap"
  ipaddr: "192.168.7.20"
masters:
  - name: "master0"
    ipaddr: "192.168.7.21"
  - name: "master1"
    ipaddr: "192.168.7.22"
  - name: "master2"
    ipaddr: "192.168.7.23"
workers:
  - name: "worker0"
    ipaddr: "192.168.7.11"
  - name: "worker1"
    ipaddr: "192.168.7.12"
```

Then you would ssh into the the bootstrap with...

```
ssh bootstrap
```
With this config, however, you can't ssh via IP address. You can just remove the file if you wish.

**Comment by rahulsingh45-tech** [comment]
[root@BACADGSL .ssh]# ssh -i /home/bacauser/.ssh/id_rsa.pub core@10.254.9.62
Enter passphrase for key '/home/bacauser/.ssh/id_rsa.pub': 
Enter passphrase for key '/home/bacauser/.ssh/id_rsa.pub': 
Enter passphrase for key '/home/bacauser/.ssh/id_rsa.pub': 
Permission denied (publickey,gssapi-keyex,gssapi-with-mic).


We have created edited config file under .ssh  directory.
"[root@BACADGSL .ssh]# cat config 
Host rhcos01
HostName localhost
Port 2222
User core
StrictHostKeyChecking no
UserKnownHostsFile /dev/null"
--------------------------------

After above changes we are getting below error while runing ssh command
 [bacauser@BACADGSL ~]$ ssh -i /home/bacauser/.ssh/id_rsa.pub core@10.254.9.62
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
@         WARNING: UNPROTECTED PRIVATE KEY FILE!          @
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
Permissions 0775 for '/home/bacauser/.ssh/id_rsa.pub' are too open.
It is required that your private key files are NOT accessible by others.
This private key will be ignored.
Load key "/home/bacauser/.ssh/id_rsa.pub": bad permissions
Permission denied (publickey,gssapi-keyex,gssapi-with-mic).

-----------------------------------------------------------

in above error we found that /home/bacauser/.ssh/id_rsa.pub bad permission. so we have revoke others permission as below.


[root@BACADGSL .ssh]# chmod 600 *
[root@BACADGSL .ssh]# ls -ltr
total 16
-rw-------. 1 bacauser bacauser 1766 Apr 30 21:34 id_rsa
-rw-------. 1 bacauser bacauser  399 Apr 30 21:34 id_rsa.pub
-rw-------. 1 root     root      106 May 14 19:24 config
-rw-------. 1 bacauser bacauser  344 May 14 19:44 known_hosts

-----------------------------------------------------------

and we tried to run ssh command getting below error. Please help to resolve the same.

[bacauser@BACADGSL ~]$ ssh -i /home/bacauser/.ssh/id_rsa.pub core@10.254.9.62
Enter passphrase for key '/home/bacauser/.ssh/id_rsa.pub': 
Enter passphrase for key '/home/bacauser/.ssh/id_rsa.pub': 
Enter passphrase for key '/home/bacauser/.ssh/id_rsa.pub': 
Permission denied (publickey,gssapi-keyex,gssapi-with-mic).

**Comment by christianh814** [comment]
You need to ssh with the private key not the public key

**Comment by rahulsingh45-tech** [comment]
Able to connect now via ssh with private key. Thanks Christian.

However , in rhcos VM we are still unable to login and also continusously getting below error that can be ignored but how can we disable logging of this error. because are not able to work on it due to this continueous error.

SELinux: mount invalid. Same superblock, different security settings for (dev mqueue, type mqueue)"

**Comment by christianh814** [comment]
Logging into the VM console is disabled on RHCOS. The only way to login is via ssh.

**Comment by rahulsingh45-tech** [comment]
That means in future , its not possible to directly login rhcos machine ? I am new to rhcos , can you please help with the further steps after booting rhocos VM. 

As per documentation , i was trying to verify bootstrap process using below command:
[core@master0 openshift]$ ./openshift-install wait-for bootstrap-complete --log-level debug
-bash: ./openshift-install: No such file or directory
[core@master0 openshift]$ openshift-install wait-for bootstrap-complete --log-level debug
-bash: openshift-install: command not found

**Comment by christianh814** [comment]
@rahulsingh45-tech 

It seems that you lack some basic troubleshooting skills. For example, this error clearly states what the issue is...

```
-bash: ./openshift-install: No such file or directory
```

Also, I'm not sure why you're running this on the master?

```
[core@master0 openshift]$ ./openshift-install wait-for bootstrap-complete --log-level debug
```

So here is some tips to help you. Learn the basics of OpenShift by going to: https://learn.openshift.com/introduction/

Also, Red Hat is offering **_FREE_** courses for a limited time. For example, this one on Containerizing applications: https://www.redhat.com/en/services/training/do080-deploying-containerized-applications-technical-overview

There are also technical overviews that might be helpful to you.

- Red Hat Enterprise Linux Technical Overview (RH024) - https://www.redhat.com/en/services/training/rh024-red-hat-linux-technical-overview
- Ansible Essentials: Simplicity in Automation Technical Overview (DO007) - https://www.redhat.com/en/services/training/do007-ansible-essentials-simplicity-automation-technical-overview
- Deploying Containerized Applications Tech Overview (DO080) - https://www.redhat.com/en/services/training/do080-deploying-containerized-applications-technical-overview
- Virtualization and Infrastructure Migration Technical Overview (RH018) - https://www.redhat.com/en/services/training/rh018-virtualization-and-infrastructure-migration-technical-overview

Once you've gotten comfortable with these; feel free to try this repo again. If you're still having issues, feel free to open up another issue (but only AFTER you've done all these courses...which are **_FREE_**)
